expression/constant.go

```diff
@@ -71,6 +71,19 @@ type Constant struct {
 	collationInfo
 }
 
+// Clone implements Expression interface.
+func (c *Constant) Clone() Expression {
```

Comment:
Why change this?

Reply:
From  #19029, I think it's the right way for Clone. We should clone the retType too. Though it's not necessary for this PR.

Reply:
If this modification is not necessary for this PR, I think we can remove it.